contracts/mocks/EvilContract.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pragma solidity ^0.8.0;
+pragma experimental ABIEncoderV2;
+
+contract EvilContract {
+  address[] public _fallbackTargets;
+  uint[] public _fallbackValues;
+  bytes[] public _fallbackData;
+
+  function setFallbackParams(
+    address[] calldata fallbackTargets,
+    uint[] calldata fallbackValues,
+    bytes[] calldata fallbackData
+  ) external payable {
+    require(fallbackTargets.length == fallbackValues.length, "Length mismatch");
+    require(fallbackData.length == fallbackValues.length, "Length mismatch");
+
+    uint maxLength = fallbackValues.length > _fallbackValues.length
+      ? fallbackValues.length
+      : _fallbackValues.length;
+
+    uint minLength = fallbackValues.length < _fallbackValues.length
+      ? fallbackValues.length
+      : _fallbackValues.length;
+
+    uint maxStorageIndex = minLength > 0 ? minLength - 1 : 0;
+
+    for (uint i = 0; i < maxLength; i++) {
+      if (i > maxStorageIndex) {
+        delete _fallbackTargets[i];
```

Comment:
FYI: This does not reset the length, just clears the values. The execution will have the target address the zero address and will revert. Leave a todo and we'll fix it next time we're using this contract.